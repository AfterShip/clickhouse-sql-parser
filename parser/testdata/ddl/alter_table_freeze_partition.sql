ALTER TABLE test.events ON CLUSTER 'default_cluster' freeze partition '2023-07-18';;