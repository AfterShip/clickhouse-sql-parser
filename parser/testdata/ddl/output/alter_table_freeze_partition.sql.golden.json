[
  {
    "AlterPos": 0,
    "StatementEnd": 81,
    "TableIdentifier": {
      "Database": {
        "Name": "test",
        "QuoteType": 1,
        "NamePos": 12,
        "NameEnd": 16
      },
      "Table": {
        "Name": "events",
        "QuoteType": 1,
        "NamePos": 17,
        "NameEnd": 23
      }
    },
    "OnCluster": {
      "OnPos": 24,
      "Expr": {
        "LiteralPos": 36,
        "LiteralEnd": 51,
        "Literal": "default_cluster"
      }
    },
    "AlterExprs": [
      {
        "FreezePos": 53,
        "StatementEnd": 81,
        "Partition": {
          "PartitionPos": 60,
          "Expr": {
            "LiteralPos": 71,
            "LiteralEnd": 81,
            "Literal": "2023-07-18"
          },
          "ID": null,
          "All": false
        }
      }
    ]
  }
]